File: Patterns/Manuscript/manuscript/code/java/DesignPatterns/src/main/java/bridge/SoundSystem.java
package bridge;

class SoundSystem implements Device {
    private boolean isOn;
    private int volume;

    public SoundSystem() {
        this.isOn = false;
        this.volume = 50; // Default volume
    }

    @Override
    public void powerOn() {
        isOn = true;
        System.out.println("Sound system powered on");
    }

    @Override
    public void powerOff() {
        isOn = false;
        System.out.println("Sound system powered off");
    }

    @Override
    public void adjustVolume(int volume) {
        this.volume = volume;
        System.out.println("Adjusting sound system volume to " + volume);
    }

    @Override
    public void changeChannel(int channel) {
        // Sound system does not have channels
        System.out.println("Sound system does not have channels");
    }
}

File: code/csharp/DesignPatterns/bridge/SoundSystem.cs
namespace Bridge;
using System;

class SoundSystem : IDevice
{
    private bool _isOn;
    private int _volume;

    public SoundSystem()
    {
        this._isOn = false;
        this._volume = 50; // Default volume
    }

    public void PowerOn()
    {
        _isOn = true;
        Console.WriteLine("Sound system powered on");
    }

    public void PowerOff()
    {
        _isOn = false;
        Console.WriteLine("Sound system powered off");
    }

    public void AdjustVolume(int volume)
    {
        this._volume = volume;
        Console.WriteLine("Adjusting sound system volume to " + volume);
    }

    public void ChangeChannel(int channel)
    {
        // Sound system does not have channels
        Console.WriteLine("Sound system does not have channels");
    }
}
